### ISSUE TITLE
The latest release 8.0.0 doesn't work on Android 11 and older versions. 

### ISSUE THREAD

**Issue by jindog** [googlesamples/android-testdpc#132]
When I download and install it on Pixel 2 with Android R (11), I can't launch TestDPC app in work profile

reproduction steps: 

1. Download and install TestDPC 8.0.0 to any device with Android R 
2. Set up work profile 
3. Try to launch TestDPC (Profile Owner) to control Work Profile 

09-13 16:02:49.818 12314 12314 E AndroidRuntime: FATAL EXCEPTION: main
09-13 16:02:49.818 12314 12314 E AndroidRuntime: Process: com.afwsamples.testdpc, PID: 12314
09-13 16:02:49.818 12314 12314 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.afwsamples.testdpc/com.afwsamples.testdpc.PolicyManagementActivity}: java.lang.IllegalStateException: requires API level S, device's on 30
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:223)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7656)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: Caused by: java.lang.IllegalStateException: requires API level S, device's on 30
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at com.afwsamples.testdpc.common.Util.requireAndroidS(Util.java:254)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at com.afwsamples.testdpc.DevicePolicyManagerGatewayImpl.isPreferentialNetworkServiceEnabled(DevicePolicyManagerGatewayImpl.java:497)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at com.afwsamples.testdpc.policy.PolicyManagementFragment.onCreatePreferences(PolicyManagementFragment.java:616)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at androidx.preference.PreferenceFragment.onCreate(PreferenceFragment.java:225)
09-13 16:02:49.818 12314 12314 E AndroidRuntime: 	at com.afwsamples.testdpc.common.BaseSearchablePolicyPreferenceFragment.onCreate(BaseSearchablePolicyPreferenceFragment.java:53)

I think, to fix this issue you need to refactor PreferentialNetwork Ui stuffs (e.g. applying TargetAPI annotation ...) 

```
===================================================================

diff --git a/app/src/main/java/com/afwsamples/testdpc/policy/PolicyManagementFragment.java b/app/src/main/java/com/afwsamples/testdpc/policy/PolicyManagementFragment.java
--- a/app/src/main/java/com/afwsamples/testdpc/policy/PolicyManagementFragment.java	(revision cf9374bc4c7d1a548bbeb8d9bd05828a9d05cd66)
+++ b/app/src/main/java/com/afwsamples/testdpc/policy/PolicyManagementFragment.java	(revision b20a357effb3a7e00f64f901d7e51171367fb4cd)
@@ -611,10 +611,7 @@
         mPreferentialNetworkServiceSwitchPreference = (SwitchPreference) findPreference(
                 SET_GET_PREFERENTIAL_NETWORK_SERVICE_STATUS);
         mPreferentialNetworkServiceSwitchPreference.setOnPreferenceChangeListener(this);
-        if (mIsProfileOwner) {
-            mPreferentialNetworkServiceSwitchPreference.setChecked(
-                    mDevicePolicyManagerGateway.isPreferentialNetworkServiceEnabled());
-        }
+
         findPreference(LOCK_SCREEN_POLICY_KEY).setOnPreferenceClickListener(this);
         findPreference(PASSWORD_CONSTRAINTS_KEY).setOnPreferenceClickListener(this);
         findPreference(RESET_PASSWORD_KEY).setOnPreferenceClickListener(this);
@@ -843,6 +840,20 @@
         reloadEnableLogoutUi();
         reloadAutoBrightnessUi();
         reloadPersonalAppsSuspendedUi();
+
+        reloadPreferentialNetworkServicesUi();
+    }
+
+    @TargetApi(VERSION_CODES.S)
+    private void reloadPreferentialNetworkServicesUi() {
+        if (Util.SDK_INT < VERSION_CODES.S) {
+            return;
+        }
+
+        if (mIsProfileOwner) {
+            mPreferentialNetworkServiceSwitchPreference.setChecked(
+                    mDevicePolicyManagerGateway.isPreferentialNetworkServiceEnabled());
+        }
     }
 
     @Override
```

**Comment by jscott1989** [comment]
Thanks for the report. I'll get this fixed asap

**Comment by jscott1989** [comment]
I've made the fix and 8.0.01 will be available soon.

The code will be updated a bit after that as we're currently reworking our integration with github to hopefully make the whole thing nicer
